### PR TITLE
Working with collections

### PR DESCRIPTION
**Pull request by ameliatalkington**
Screen Shot 2021-02-04 at 10 40 30 AM

### CODE REVIEW

File: working-with-collections/main.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        count += 11;
+      } else if (objectOfCards[keys] === 'jack' || objectOfCards[keys] === 'queen' || objectOfCards[keys] === 'king') {
+        count += 10;
+      } else {
+        count += objectOfCards[keys];
+      }
+    }
+  }
+  return count;
+}
```
Comment (thebearingedge) [inline working-with-collections/main.js:60]: This is the only part of your algorithm I'm having a hard time understanding. What is `arrayOfObjects`? Is it an array of card objects? Maybe the variable name could be improved.

Also, `objectOfCards`. Is it a single card object?

the inner `for ... in` loop can be omitted if `objectOfCards` is a card. You can access `card.rank` directly without a `for ... in`.

**Review by thebearingedge — CHANGES_REQUESTED** [review]
Great job, but some of the code could be clearer.

**Comment by ameliatalkington** [comment]
Screen Shot 2021-02-04 at 4 40 09 PM

File: working-with-collections/main.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      var cardObject = {};
+      cardObject.suit = suit[j];
+      cardObject.rank = rank[k];
+      arrayOfCardObjects.push(cardObject);
+    }
+  }
+  return arrayOfCardObjects;
+}
+
+function calculateScore(shuffledDeck) {
```
Comment (thebearingedge) [inline working-with-collections/main.js:48]: This parameter name is misleading. It's not going to be the whole `shuffledDeck`, it's one of the player's `hand`s.

```js
function calculateScore(hand) { // 👀 

}
```

**Review by thebearingedge — CHANGES_REQUESTED** [review]
Much, much better. Good work! There is one more adjustment you can make that will improve the readability of your code.

**Comment by ameliatalkington** [comment]
Screen Shot 2021-02-05 at 12 07 31 PM

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**